Bio/PDB/mmtf/mmtfio.py

```diff
@@ -100,6 +100,7 @@ def _save_structure(self, filepath, select):
         header_dict = defaultdict(str, self.structure.header)
         if header_dict["resolution"] == "":
```

Comment:
With the `defaultdict`, this key access will automatically create the key if it does not exist. I'd simply check with `dict.get` instead, like this:

```python
if header_dict.get("resolution") is None:  # does not create key
    header_dict["resolution"] = None
...
```

Reply:
Yes - that surprising behaviour of the defaultdict caused me a major memory leak once!